include/pagmo/population.hpp

```diff
@@ -411,6 +432,23 @@ class population
     }
 
 private:
+    // Short routine tp update the champion. Does nothing if the problem is MO
+    inline void update_champion(const vector_double& x, const vector_double& f)
+    {
+        assert(f.size() > 0u);
+        // If the problem has multiple objectives do nothing
+        if (m_prob.get_nobj() == 1u) {
+            // If the champion does not exist create it
+            if (m_champion_x.size() == 0u) {
+                m_champion_x = x;
+                m_champion_f = f;
+            }
```

Comment:
Wouldn't here be better to use an ``else if`` instead, so we avoid re-comparing the fitness in case this is the first champion being set?